client/trino-jdbc/src/main/java/io/trino/jdbc/NonRegisteringTrinoDriver.java

```diff
@@ -87,7 +107,22 @@ public boolean acceptsURL(String url)
     @Override
     public DriverPropertyInfo[] getPropertyInfo(String url, Properties info)
     {
-        return TrinoDriverUri.getPropertyInfo(url, info);
+        Properties properties = urlProperties(url, info);
+        return allProperties().stream()
+                .filter(property -> property.isValid(properties))
```

Comment:
Why is it ok to filter out invalid properties here, instead of throwing an exception? Add a code comment.